flair/models/entity_mention_linking.py

```diff
@@ -0,0 +1,1174 @@
+                    entity_type = PRETRAINED_HYBRID_MODELS[model_name_or_path]
+
+        else:
+            if isinstance(model_name_or_path, str):
+                model_name_or_path = cast(str, model_name_or_path)
+                if model_name_or_path in ENTITY_TYPES:
+                    model_name_or_path = ENTITY_TYPE_TO_DENSE_MODEL[model_name_or_path]
+
+        assert (
+            entity_type is not None
+        ), f"Impossible to determine entity type for model `{model_name_or_path}`: please specify via `entity_type`"
+
+        return model_name_or_path, entity_type
+
+    @staticmethod
+    def __get_dictionary_path(
+        model_name_or_path: str,
+        dictionary_name_or_path: Optional[Union[str, Path]] = None,
+    ) -> Union[str, Path]:
+        """Try to figure out what dictionary (depending on the model) the user wants."""
+        if model_name_or_path in STRING_MATCHING_MODELS and dictionary_name_or_path is None:
+            raise ValueError(
+                "When using a string-matching candidate generator you must specify `dictionary_name_or_path`!"
+            )
+
+        if dictionary_name_or_path is not None and isinstance(dictionary_name_or_path, str):
+            dictionary_name_or_path = cast(str, dictionary_name_or_path)
+
+            if dictionary_name_or_path in ENTITY_TYPES:
+                dictionary_name_or_path = ENTITY_TYPE_TO_DICTIONARY[dictionary_name_or_path]
+        else:
+            if model_name_or_path in MODEL_NAME_TO_DICTIONARY:
+                dictionary_name_or_path = MODEL_NAME_TO_DICTIONARY[model_name_or_path]
+            elif model_name_or_path in ENTITY_TYPE_TO_DICTIONARY:
+                dictionary_name_or_path = ENTITY_TYPE_TO_DICTIONARY[model_name_or_path]
+            else:
+                raise ValueError(
+                    f"When using a custom model you need to specify a dictionary. Available options are: {list(ENTITY_TYPE_TO_DICTIONARY.values())}. "
+                    "Or provide a path to a dictionary file."
+                )
+
+        return dictionary_name_or_path
+
+    def forward_loss(self, data_points: List[DT]) -> Tuple[torch.Tensor, int]:
+        raise NotImplementedError("The EntityLinker cannot be trained")
+
+    @classmethod
+    def load(cls, model_path: Union[str, Path, Dict[str, Any]]) -> "EntityMentionLinker":
+        from typing import cast
+
+        return cast("EntityMentionLinker", super().load(model_path=model_path))
+
+    def evaluate(
+        self,
+        data_points: Union[List[Sentence], Dataset],
+        gold_label_type: str,
+        out_path: Optional[Union[str, Path]] = None,
+        embedding_storage_mode: str = "none",
+        mini_batch_size: int = 32,
+        main_evaluation_metric: Tuple[str, str] = ("accuracy", "f1-score"),
+        exclude_labels: List[str] = [],
+        gold_label_dictionary: Optional[Dictionary] = None,
+        return_loss: bool = True,
+        k: int = 1,
```

Comment:
Many of these parameters are unused (like `out_path`, `embedding_storage_mode`, etc.). This is a limitation of our current evaluate signature of the `Model` class. 

Some possibilities (not in this PR):
- Have the EntityMentionLinker inherit from Classifier instead of Model and reuse its evaluate function. This would also entail adapting the predict method and determining how many of these parameters make sense for an untrained model. For instance, is there a batch size?
- Implement functionality for as many parameters as possible. For instance, we could implement something for out_path.